### DISCUSSION TITLE
images are getting cropped

### DISCUSSION THREAD

**Discussion by scarf005** [lowlighter/metrics#787] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/54838975/149890111-d13d59eb-1531-462f-bf5b-6a26346f5f75.png)

actions file: [yaml](https://github.com/scarf005/scarf005/blob/d7669b650fba0b22b9de5ce1f701e1bf7ef7ed08/.github/workflows/main.yaml)
maybe it's related to plugin order?

**Comment by lowlighter** [comment, marked as answer]
I think it's related to the activity plugin, stuff after it has an "indent" while it shouldn't so they end up getting shifted to the right outside the svg viewbox.

I'm not sure yet, but I think it's caused by one of your commit message not properly handled:
- some of them are longer than expected and should be cut with an ellipsis (...) at the end
- since you use `>` characters, maybe it's not properly escaped which break the html by closing a tag or something like this

<img src="https://user-images.githubusercontent.com/22963968/149943225-a89b575c-7a9b-4bd2-b03d-7088d7b5f09e.png">

**Reply by lowlighter** [reply]
The offset issue should have been fixed now 👍

**Comment by njzjz** [comment]
Got a similar issue. My config is [here](https://github.com/deepmodeling/.github/blob/82f3a8f1ab16c328c24c3c9fb820655d9737de50/.github/workflows/metrics.yml).

![image](https://user-images.githubusercontent.com/9496702/150234879-fcb59eef-1b74-47f7-a5b1-aa3ee6129a5c.png)

Note: I do not enable the activity plugin.

**Reply by lowlighter** [reply]
The vertical cropping issue can actually be solved by using the `config_padding` option. Because the SVG render ultimately depends on the OS/browser/fonts combination, it can result in slightly small changes in the height across different platforms.

To mitigate this, the `config_padding` already increase the height by 8px + 11%  of its detected size by the puppeteer instance, but in your case it doesn't seem enough. You could try using higher values, like `0, 8 + 13%` for example 

Here's the full explanation of the option:
https://github.com/lowlighter/metrics/blob/0c4ce7ce02e34cc5a9d1654461c69c6b2890dce6/source/plugins/core/metadata.yml#L255-L273

Hope this helps 🙂 !
